### PR TITLE
Updated "What's New" with my recent commits.

### PR DESCRIPTION
**Pull request by larryhastings**

### CODE REVIEW

**Comment by larryhastings** [comment]
Pablo: I tagged you because I made two edits not in my section.  ISTM the document was using "annotations" where it meant "type hints" in a couple places.  Do you agree, or should I back out of those changes?

File: Doc/whatsnew/3.10.rst [hunk 1]
```diff
@@ -987,10 +991,18 @@ inspect
...
-local and global namespaces.
-(Contributed by Batuhan Taskaya in :issue:`41960`.)
+Added :func:`inspect.get_annotations`, which safely computes the annotations
+defined on an object. It works around the quirks of accessing the annotations
+on various types of objects, and makes very few assumptions about the object
+it examines. :func:`inspect.get_annotations` can also correctly un-stringize
+stringized annotations.  :func:`inspect.get_annotations` is now considered
+best practice for accessing the annotations dict defined on any Python object.
+Relatedly, :func:`inspect.signature`,
+:func:`inspect.from_callable`, and `inspect.from_function` now call
```
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.10.rst]: Tests are failing due to this line:
```suggestion
:func:`inspect.from_callable`, and :func:`inspect.from_function` now call
```
alternatively you can just use double backticks.

File: Doc/whatsnew/3.10.rst [hunk 2]
```diff
@@ -659,10 +659,10 @@ are added to enable the warning.
 See :ref:`io-text-encoding` for more information.
 
 
-New Features Related to Type Annotations
```
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.10.rst:662]: Personally, I'd rather use "annotations" here if you plan to add your entry under this section. ``Class and module objects now lazy-create empty annotations dicts on demand`` seems something related to annotations, not just type hinting ;). Or maybe we can just write both: ``New Features Related to Type Annotations and Hints``. What do you think?

**Review by Fidget-Spinner — COMMENTED** [review]
Larry, thanks for the patch. When I wrote the sections on type hints, I was using type annotations and hints interchangeably. I forgot that there are subtle differences between the two. Most of the changes LGTM. Thanks for making them!

File: Doc/whatsnew/3.10.rst (hunk 1 above)
Comment (larryhastings) [reply Doc/whatsnew/3.10.rst]: I'm a big fan of `:func:` these days, definitely going that way with it.

Comment (larryhastings) [reply Doc/whatsnew/3.10.rst]: And, maybe I've been thinking about this topic way too long, but I don't think the difference is subtle ;-)

Think of it this way: an annotation is defined by a syntactic context, a type hint is defined as something inheriting from `TypeVar` (or its stringized equivalent, etc etc).  It's true that lots of people only ever use type hints as annotations, and only ever annotate with type hints.  But they're still very different concepts--confusing the two is like confusing a function argument and an int.

Comment (Fidget-Spinner) [reply Doc/whatsnew/3.10.rst]: Yeah I agree with what you said - type hints need not be annotations, it just needs to bear some meaning to the type checker (eg. type comments). Whilst annotations is a purely syntactical construct (well bar some exceptions like in ``dataclasses`` and runtime checking)

I only noticed the major difference *while* I was writing another comment and reflecting for the 10 seconds after posting 😉 . Hence why I promptly deleted that comment after feeling pretty dumb hehe.

PS. regarding `

> a type hint is defined as something inheriting from TypeVar

Where can I read more about this? I didn't know and I can't find much info about that.

**Comment by larryhastings** [comment]
I fixed the bugs Fidget-Spinner pointed out (I think?), and made a couple more clarifications / edits to _What's New In Python 3.10._

I wonder why I don't see the test failures on my machine--`make html` and `make suspicious` both run fine, without errors.  Is there another command I should be running to reproduce these errors?  Is my doc build environment possibly out of date?

File: Doc/whatsnew/3.10.rst (hunk 1 above)
Comment (larryhastings) [reply Doc/whatsnew/3.10.rst]: Actually I misspoke.  There are a couple other things that can be type hints.  Types, for example :D

I'm not an expert on `typing` or type hints.  All I can really tell you is that `typing._type_check()` appears to be as canonical a source as one is likely to find.  It smiles upon instances of `type`,  `Any`, `NoReturn`,  `TypeVar`, `ForwardRef`, `Union`, `ParamSpec`, and some instances of `_GenericAlias`.  It complains about instances of `_SpecialForm`, the objects `Generic` and `Protocol`, or any callable not previously explicitly permitted.  And anything I didn't mention above seems to get a pass, too.

**Comment by Fidget-Spinner** [comment]
> Is there another command I should be running to reproduce these errors? 

The checks also run ``python3 tools/rstlint.py -i tools -i ./venv -i README.rst``which is located at ``Doc/tools/rstlint.py``. Any warning it emits will cause the bots to error.

Here's a copy of the command the bot runs:
``make check html suspicious SPHINXOPTS="-q -W -j4"``

So I think if you run ``make check suspicious`` , you should get the same result.

File: Doc/whatsnew/3.10.rst [hunk 3]
```diff
@@ -733,8 +733,8 @@ especially when forward references or invalid types were involved. Compare::
    StrCache = 'Cache[str]'  # a type alias
    LOG_PREFIX = 'LOG[DEBUG]'  # a module constant
 
-Now the :mod:`typing` module has a special annotation :data:`TypeAlias` to
-declare type aliases more explicitly::
+Now the :mod:`typing` module has a special value :data:`TypeAlias`
```
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.10.rst:736]: Really minor nit (feel free to ignore it):
```suggestion
Now the :mod:`typing` module has a special type :data:`TypeAlias`
```

**Review by Fidget-Spinner — APPROVED** [review]
Thanks!

File: Doc/whatsnew/3.10.rst (hunk 3 above)
Comment (larryhastings) [reply Doc/whatsnew/3.10.rst:736]: TypeAlias is a function, not a type.  Please see `Lib/typing.py` line 533.

**Comment by larryhastings** [comment]
Thanks for the review and the feedback!

File: Doc/whatsnew/3.10.rst (hunk 3 above)
Comment (Fidget-Spinner) [reply Doc/whatsnew/3.10.rst:736]: I think in ``typing.py`` , the term ``type`` isn't strictly referring to what Python normally deems a ``type`` (it's kind of confusing in some places) but rather what the type checker deems as a type.

Ref: ``Lib/typing.py`` line 449. ``Union`` is called a "type" in the docs, but it's created the same way ``TypeAlias`` is (via a function decorated with the ``_SpecialForm`` class decorator). Another similar example is line 379 ``Any``, which also uses a decorated function, but is called a "type".

Even then I think you're right here. ``TypeAlias`` isn't really a type, it's considered a special marker/construct. So I'm just plain bike shedding here. Sorry :P, I just wanted to share something I found pretty interesting.

**Pull request merged**